packages/actions-spec/index.d.ts

```diff
@@ -88,6 +94,35 @@ export interface ActionParameter {
   required?: boolean;
 }
 
+/**
+ * Input field type to present to the user. Normally resembling the respective
+ * [HTML `input` types](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input)
+ * or standard HTML element (e.g. `select`) for the platform being used (web, mobile, etc).
+ * @default `text`
+ */
+export type ActionParameterType =
```

Comment:
I'd like to suggest that we type this a bit more strictly:

```ts
export type GeneralParameterType =
  | 'text'
  | 'email'
  | 'url'
  | 'number'
  | 'date'
  | 'datetime-local'
  | 'textarea';

export type SelectableParameterType = 'select' | 'radio' | 'checkbox';

export type ParameterType = GeneralParameterType | SelectableParameterType;

export interface Parameter<T extends ParameterType> {
  /** input field type */
  type?: T;
  /** regular expression pattern to validate user input client side */
  pattern?: string;
  /** human-readable description of the `pattern` */
  patternDescription?: string;
  /** parameter name in url */
  name: string;
  /** input placeholder */
  label?: string;
  /** input required */
  required?: boolean;
}

export interface ParameterSelectable<T extends ParameterType>
  extends Parameter<T> {
  pattern?: never;
  patternDescription?: never;
  options: Array<{
    /** displayed UI label of this selectable option */
    label: string;
    /** value of this selectable option */
    value: string;
    /** whether this option should be selected by default */
    selected?: boolean;
  }>;
}

export type TypedParameter<T extends ParameterType = ParameterType> =
  T extends SelectableParameterType ? ParameterSelectable<T> : Parameter<T>;
```

that way, only `select`, `checkbox` and `radio` would have `options` and should never receive `pattern` and `patternDescription`

Can be verified like so:
```ts
const test: TypedParameter = {}; // ignore ts error here

if (test.type === 'select') {
  test.options; // no error
}

if (test.type === 'text') {
  test.options; // TS2339: Property options does not exist on type Parameter<"text">
}
```

Reply:
def agree with Filipp. More stricter types we have - more errors we can eliminate at build time, and avoid extra support questions

Reply:
Love this! A great addition and will help improve the DX of using the types

Reply:
I just pushed the updates and update the readme in a few places. I added a statement noting that the code blocks in the readme may be the simplified types and the actual ones in the `index.d.ts` may be more complicated for better type safety and DX